File: C++/Level.2/카펫.cpp
#include <string>
#include <vector>

using namespace std;


vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    for(int row = 1; row * row <= yellow; ++row) {
        if(yellow % row == 0) {
            int column = yellow / row;
            if(brown == ((row+2) * (column+2) - yellow))
            {
                answer.push_back(column+2);
                answer.push_back(row+2);
                break;
            }
        }
    }
    return answer;
}

File: JavaScript/Level.2/카펫.js
function solution(brown, yellow) {
    var answer = [];
    let total = brown + yellow;
    let sqrt = Math.floor(Math.sqrt(total));
    for (let column = 3; column <= sqrt; ++column) {
        if (total % column === 0) {
            let row = total / column;
            if ((row - 2) * (column - 2) === yellow) {
                answer = [row, column];
                break;
            }
        }
    }
    return answer;
}
